src/snowflake/connector/pandas_tools.py

```diff
@@ -352,9 +352,15 @@ def write_pandas(
         # if the column name contains a double quote, we need to escape it by replacing with two double quotes
         # https://docs.snowflake.com/en/sql-reference/identifiers-syntax#double-quoted-identifiers
         snowflake_column_names = [str(c).replace('"', '""') for c in df.columns]
+        tz_columns = [
+            str(c).replace('"', '""') for c in df.columns if pandas.api.types.is_datetime64tz_dtype(df[c])
+        ]
```

Comment:
can you explain why this change is necessary?

Reply:
Same reason as explained in the comment starting at line 352

 https://github.com/snowflakedb/snowflake-connector-python/blob/92808e5dd482f7b17a23293d56aea2d5076ae5b0/src/snowflake/connector/pandas_tools.py#L352